### PR TITLE
Feature/itinerary chat

### PR DESCRIPTION
**Pull request by 276746**
Add really simple UI for chat for futur prompt to create itineraries

(Shipped with tests)

Screenshot 2024-04-25 224947

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            onTextChange = { textState = it },
+            onSend = {
+              if (textState.isNotBlank()) {
+                messages.add(textState)
+                textState = ""
+                coroutineScope.launch { listState.animateScrollToItem(messages.size - 1) }
+              }
+            })
+      },
+      containerColor = Color.White) { innerPadding ->
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt:65]: Should change this to match the app's color scheme. Try `MaterialTheme.colorScheme.PrimaryContainer` (or something similar)

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+@Composable
+fun MessageBubble(message: String, isUserMessage: Boolean) {
+  Row(
+      modifier = Modifier.fillMaxWidth().padding(horizontal = 8.dp, vertical = 4.dp),
+      horizontalArrangement = if (isUserMessage) Arrangement.End else Arrangement.Start) {
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier =
+                Modifier.background(
+                        color = if (isUserMessage) Color(0xFFDCF8C6) else Color.LightGray,
```
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt:97]: Ditto here, try not to use hardcoded color values.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt
Comment (ethangraham2001) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt:1]: lgtm just get rid of the hardcoded colors when you get a chance

File: app/src/test/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChatTest.kt
Comment (ethangraham2001) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChatTest.kt:1]: Tests are looking good. I think sonar is pissy about the % of coverage though

**Review by ethangraham2001 — COMMENTED** [review]
Looks good. Just replace hardcoded colors with `MaterialTheme` and up the coverage a bit when you get a chance. Not top priority for this week

**Comment by 276746** [comment]
Will merge on main

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt (hunk 1 above)
Comment (276746) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt:65]: noted will change

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt (hunk 2 above)
Comment (276746) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryChat.kt:97]: same, will change

**Pull request closed without merging**